dts/arm/nordic/nrf5340_cpu0_common.dtsi

```diff
@@ -0,0 +1,328 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+flash_controller: flash-controller@39000 {
+	compatible = "nordic,nrf53-flash-controller";
+	reg = <0x39000 0x1000>;
+
+	#address-cells = <1>;
+	#size-cells = <1>;
+
+	label = "NRF_FLASH_DRV_NAME";
+
+	flash0: flash@0 {
+		compatible = "soc-nv-flash";
+		label = "NRF_FLASH";
+		erase-block-size = <4096>;
+		write-block-size = <4>;
+	};
+};
+
+adc: adc@e000 {
+	compatible = "nordic,nrf-saadc";
+	reg = <0xe000 0x1000>;
+	interrupts = <14 1>;
+	status = "disabled";
+	label = "ADC_0";
+	#io-channel-cells = <1>;
+};
+
+dppic: dppic@17000 {
+	compatible = "nordic,nrf-dppic";
+	reg = <0x17000 0x1000>;
+	status = "okay";
+	label = "DPPIC";
+};
+
+egu0: egu@1b000 {
+	compatible = "nordic,nrf-egu";
+	reg = <0x1b000 0x1000>;
+	interrupts = <27 1>;
+	status = "okay";
+};
+
+egu1: egu@1c000 {
+	compatible = "nordic,nrf-egu";
+	reg = <0x1c000 0x1000>;
+	interrupts = <28 1>;
+	status = "okay";
+};
+
+egu2: egu@1d000 {
+	compatible = "nordic,nrf-egu";
+	reg = <0x1d000 0x1000>;
+	interrupts = <29 1>;
+	status = "okay";
+};
+
+egu3: egu@1e000 {
+	compatible = "nordic,nrf-egu";
+	reg = <0x1e000 0x1000>;
+	interrupts = <30 1>;
+	status = "okay";
+};
+
+egu4: egu@1f000 {
+	compatible = "nordic,nrf-egu";
+	reg = <0x1f000 0x1000>;
+	interrupts = <31 1>;
+	status = "okay";
+};
+
+egu5: egu@20000 {
+	compatible = "nordic,nrf-egu";
+	reg = <0x20000 0x1000>;
+	interrupts = <32 1>;
+	status = "okay";
+};
+
+i2s0: i2s@28000 {
+	compatible = "nordic,nrf-i2s";
+	#address-cells = <1>;
+	#size-cells = <0>;
+	reg = <0x28000 0x1000>;
+	interrupts = <40 1>;
+	status = "disabled";
+	label = "I2S_0";
+};
+
+ipc: ipc@2a000 {
+	compatible = "nordic,nrf-ipc";
+	reg = <0x2a000 0x1000>;
+	interrupts = <42 1>;
+	status = "okay";
+	label = "IPC";
+};
+
+kmu: kmu@39000 {
+	compatible = "nordic,nrf-kmu";
+	reg = <0x39000 0x1000>;
+	interrupts = <57 1>;
+	status = "okay";
+};
+
+pdm0: pdm@26000 {
+	compatible = "nordic,nrf-pdm";
+	reg = <0x26000 0x1000>;
+	interrupts = <38 1>;
+	status = "disabled";
+	label = "PDM_0";
+};
+
+regulators: regulator@4000 {
+	compatible = "nordic,nrf-regulators";
+	reg = <0x4000 0x1000>;
+	status = "okay";
+};
+
+vmc: vmc@81000 {
+	compatible = "nordic,nrf-vmc";
+	reg = <0x81000 0x1000>;
+	status = "okay";
+};
+
+uart0: uart@8000 {
+	compatible = "nordic,nrf-uarte";
+	reg = <0x8000 0x1000>;
+	interrupts = <8 1>;
+	status = "disabled";
+	label = "UART_0";
+};
+
+uart1: uart@9000 {
+	compatible = "nordic,nrf-uarte";
+	reg = <0x9000 0x1000>;
+	interrupts = <9 1>;
+	status = "disabled";
+	label = "UART_1";
+};
+
+i2c0: i2c@8000 {
+	/*
+	 * This i2c node can be either TWIM or TWIS, for the user to pick:
+	 * compatible = "nordic,nrf-twim" or
+	 *              "nordic,nrf-twis".
+	 */
+	compatible = "nordic,nrf-i2c";
```

Comment:
```suggestion
```
This property is to be chosen at some further level, in a board definition, for instance.